xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/macro/AbstractMacro.java

```diff
@@ -281,4 +281,32 @@ protected MetaData getNonGeneratedContentMetaData()
         metaData.addMetaData(MetaData.NON_GENERATED_CONTENT, converted);
         return metaData;
     }
+
+    /**
+     * Helper to get the proper metadata for non generated content (i.e. content that has not gone through a
+     * Transformation) for a specific parameter. This content can be used for inline editing.
+     *
+     * @param parameterName the name of the parameter as defined in the macro
+     * @return the new metadata with the content type for the content represented as a string (e.g.
+     *         {@code java.util.List< org.xwiki.rendering.block.Block >} for content of type {@code List<Block>}
+     * @since 11.1RC1
+     */
+    @Unstable
+    protected MetaData getNonGeneratedContentMetaData(String parameterName)
+    {
+        MetaData metaData = new MetaData();
+        Type contentType;
+
+        if (this.contentDescriptor != null) {
+            contentType = this.contentDescriptor.getType();
+        } else {
+            contentType = DefaultContentDescriptor.DEFAULT_CONTENT_TYPE;
```

Comment:
This is the part that I don't like: you can't control the type used for macro parameters as you can for macro content. Shouldn't the type be taken from the macro parameter descriptor? Parameters already have a type specified there.

Reply:
Hm indeed you're right I shouldn't rely on content stuff here. So, just to be sure we're on the same page, you suggest I replace the whole if with something like:

```java
if (this.macroDescriptor != null) {
   contentType = this.macroDescriptor
         .getParameterDescriptorMap()
         .get(parameterName)
         .getParameterType();
} else {
   contentType = DefaultParameterDescriptor.DEFAULT_PARAMETER_TYPE;
}
```

So now the question is what would be the value of `DefaultParameterDescriptor.DEFAULT_PARAMETER_TYPE`?

Reply:
> you suggest I replace the whole if

Only the ``else`` actually. But note that the parameter descriptor has both ``getParameterType()`` and ``getDisplayType()``. Shouldn't you use the later?

Regarding ``DefaultParameterDescriptor.DEFAULT_PARAMETER_TYPE`` I would be OK to keep ``DefaultContentDescriptor.DEFAULT_CONTENT_TYPE``.

Reply:
> Shouldn't you use the later?

If I follow it's the type given by `@PropertyDisplayType`, I'd need to investigate a bit more on that one I didn't followed the discussions there.